src/Workspaces/CSharp/Portable/Simplification/CSharpNameReducer.Rewriter.cs

```diff
@@ -19,160 +19,72 @@ public Rewriter(OptionSet optionSet, CancellationToken cancellationToken)
             {
             }
 
-            public override SyntaxNode VisitPredefinedType(PredefinedTypeSyntax node)
+            private SyntaxNode SimplifyVisit<TExpression>(
```

Comment:
I appreciate the simplification.  However, I worry about all the delegate creations you're now causing.  can you capture the delegates in the constructor and then pass them in accordingly?

Remember, we're going to walk huge trees.  Causing a lambda allocation on every identifier (for example) is not good.